File: js/bomb-system.js
// bomb-system.js — 全屏清場技（炸彈）
(function() {
  window.SG = window.SG || {};

  var BOMB_CD = 120; // 120秒充能
  var FLASH_DURATION = 0.4;

  function BombSystem() {
    this.timer = BOMB_CD;
    this.ready = false;
    this.flash = 0; // 爆炸閃爍倒數
  }

  BombSystem.prototype.update = function(dt) {
    if (!this.ready) {
      this.timer -= dt;
      if (this.timer <= 0) { this.ready = true; this.timer = 0; }
    }
    if (this.flash > 0) this.flash -= dt;
  };

  BombSystem.prototype.activate = function(enemies) {
    if (!this.ready) return [];
    this.ready = false;
    this.timer = BOMB_CD;
    this.flash = FLASH_DURATION;
    // 清除所有敵人
    var killed = [];
    for (var i = enemies.length - 1; i >= 0; i--) {
      killed.push(enemies[i]);
    }
    return killed;
  };

  BombSystem.prototype.getProgress = function() {
    if (this.ready) return 1;
    return 1 - (this.timer / BOMB_CD);
  };

  BombSystem.prototype.isFlashing = function() {
    return this.flash > 0;
  };

  SG.BombSystem = BombSystem;
})();

File: unity-project/Assets/Scripts/Progression/BombSystem.cs
using System.Collections.Generic;

/// <summary>
/// BombSystem — 大招/炸彈系統
/// 對應原始架構: bomb-system.js BombSystem class
/// 累積擊殺充能，滿後可全螢幕清場
/// </summary>
public class BombSystem
{
    private const int KILLS_TO_CHARGE = 30;
    private const float RECHARGE_CD = 120f;

    private float _charge;        // 0 ~ 1
    private bool _ready;

    /// <summary>
    /// 增加充能值
    /// </summary>
    public void AddCharge(int kills)
    {
        if (_ready) return;

        _charge += (float)kills / KILLS_TO_CHARGE;
        if (_charge >= 1f)
        {
            _charge = 1f;
            _ready = true;
        }
    }

    /// <summary>
    /// 是否可施放
    /// </summary>
    public bool IsReady() => _ready;

    /// <summary>
    /// 取得充能百分比
    /// </summary>
    public float GetChargePercent() => _charge;

    /// <summary>
    /// 全螢幕清場，回傳擊殺列表
    /// </summary>
    public List<EnemyBase> Activate(List<EnemyBase> enemies)
    {
        if (!_ready) return new List<EnemyBase>();

        var killed = new List<EnemyBase>(enemies);
        foreach (var enemy in killed)
        {
            enemy.TakeDamage(float.MaxValue);
        }

        _charge = 0f;
        _ready = false;
        return killed;
    }
}
